### PR TITLE
Support version 0.4 and actix-3

### PR DESCRIPTION
**Pull request by omid**

### CODE REVIEW

**Comment by omid** [comment]
Will fix #222

File: book/actix-plugin.md [hunk 1]
```diff
@@ -8,10 +8,10 @@ Let's start with a simple actix-web application. It has `actix-web` and `serde`
 # [package] ignored for brevity
 
 [dependencies]
-actix-web = "2.0"
+actix-web = "3.0"
```
Comment (wafflespeanut) [inline book/actix-plugin.md:11]: Can you leave a comment here mentioning that 2.0 is also supported?

File: book/actix-plugin.md [hunk 2]
```diff
@@ -30,10 +30,15 @@ struct Pet {
 }
 
 async fn echo_pet(body: Json<Pet>) -> Result<Json<Pet>, ()> {
-    Ok(body)
+    let new_pet = Pet {
+        name: format!("The New {}", body.name),
+        id: body.id
+    };
```
Comment (wafflespeanut) [inline book/actix-plugin.md:36]: This really feels like an unnecessary change when we're trying to demo the plugin.

**Review by wafflespeanut — COMMENTED** [review]
Two nits.

File: book/actix-plugin.md (hunk 2 above)
Comment (omid) [reply book/actix-plugin.md:36]: I wanted to show it that the response should be a JSON for sure! (Not a HttpResponse or ...) 
And how to create it!

Or I'm wrong.

File: book/actix-plugin.md (hunk 1 above)
Comment (omid) [reply book/actix-plugin.md:12]: done, can you check plz

File: book/actix-plugin.md (hunk 2 above)
Comment (wafflespeanut) [reply book/actix-plugin.md:37]: No, it's just that it's an echo handler, so why not respond with the body as it is?

File: book/actix-plugin.md (hunk 1 above)
Comment (wafflespeanut) [reply book/actix-plugin.md:12]: We can just leave it at, `actix-web 2.0 is also supported`

File: book/actix-plugin.md (hunk 2 above)
Comment (omid) [reply book/actix-plugin.md]: OK, done

File: book/actix-plugin.md (hunk 1 above)
Comment (omid) [reply book/actix-plugin.md:12]: OK, done

**Review by wafflespeanut — APPROVED** [review]
Thank you! :smile:

**Comment by dunnock** [comment]
Hi, by some reason after this fix I can't make paperclip to pick actix-web 2.0, event after forcing that explicitly in my repo:
```
actix-web = { version = "2.0.0", default_features = false, optional = true }
```

I got in the cargo tree:
```
db v0.1.0 (/root/project/db)
├── actix-web v2.0.0
│   ├── actix-codec v0.2.0 (*)
│   ├── actix-http v1.0.1 (*)
│   ├── actix-macros v0.1.2 (*)
│   ├── actix-router v0.2.5
...
├── paperclip v0.4.2 (https://github.com/tari-labs/paperclip?branch=taris-v2#3ee6ecfc)
│   ├── anyhow v1.0.32
│   ├── itertools v0.9.0
│   │   └── either v1.6.1
│   ├── once_cell v1.4.1
│   ├── paperclip-actix v0.2.2 (https://github.com/tari-labs/paperclip?branch=taris-v2#3ee6ecfc)
│   │   ├── actix-service v1.0.6 (*)
│   │   ├── actix-web v3.0.2
```

Can you pls advice on how to force it to actix-web 2.0?

**Comment by wafflespeanut** [comment]
It's not this commit, because it doesn't affect the code. It could be a regression from #218. Either way, I can reproduce this locally. Cargo probably couldn't reuse actix-web, maybe because of the diff of dependencies between actix-web 2 and 3. There are two ways to approach this. Either we find that dependency which is causing the problem and fix it in the tree (I can't guarantee about this though) or we can expose features like `actix2` and `actix3` with `actix` defaulting to the latest version.

**Comment by dunnock** [comment]
Weird, I have amended version in a fork to depend on actix2 and it works nice. I guess if you would used some feature which is unique to actix3 that would not worked.

As to solution - I resolved that in our fork, and I guess most of people will upgrade to actix3. But I guess correct approach would be to keep 0.4 with actix2 and release 0.5 with actix3..?

**Comment by dunnock** [comment]
Yeah and I did not mention how did I noticed that - actually we use `nightly-06-30` toolchain and actix3 brings time-0.2 which does not compile on that nightly, so actix-web3 might be considered as a breaking change for us https://github.com/time-rs/time/issues/281

**Comment by wafflespeanut** [comment]
> I guess if you would used some feature which is unique to actix3 that would not worked.

But we haven't... We've disabled the default features.

> I guess correct approach would be to keep 0.4 with actix2 and release 0.5 with actix3..?

This will be a maintenance problem for us, because we'd still have to support actix-web 2.0 in the long run (at least, as long as it's maintained by actix community), which means any bug fixes will require a release in `0.4.x` in parallel with other releases. If we had features instead, we don't have to maintain multiple versions of paperclip and we can rely on everyone to switch to the latest version eventually.

**Comment by dunnock** [comment]
I see your point ref maintenance, was not aware of necessity of support for actix2, yeah features definitely make sense now. Though I guess since it is breaking change, considering actix3 would be the default that should be paperclip 0.5? Since least that would happen to people using actix2 if they autoupgrade is that their API definitions will be compiled for actix3 not compatible with actix2, but in our case it has broke build.

**Comment by wafflespeanut** [comment]
@dunnock I've now bumped minor version instead of patch version. Folks who use actix-web 2.0 can use `actix2` feature. The changes are in master now. Can you check?
